### PR TITLE
Code cleanup for bundle.py

### PR DESCRIPTION
**Pull request by wch**
This PR make some changes to bundle.py which eliminates all the errors flagged by Pyright in bundle.py.

### CODE REVIEW

File: rsconnect/bundle.py [hunk 1]
```diff
@@ -264,17 +264,27 @@ def discard_from_buffer(self, key: str):
...
+        new_data_files: dict[str, ManifestDataFile] = {}
+        deploy_dir: str
+
+        entrypoint = self.check_and_get_entrypoint()
+        if self.deploy_dir is not None:
+            deploy_dir = self.deploy_dir
+        elif entrypoint is not None and isfile(entrypoint):
+            deploy_dir = dirname(entrypoint)
+        else:
+            deploy_dir = entrypoint
```
Comment (wch) [inline rsconnect/bundle.py:286]: Add a comment that this might actually be an error case.

File: rsconnect/bundle.py [hunk 2]
```diff
@@ -264,17 +264,27 @@ def discard_from_buffer(self, key: str):
...
-        return self
+        return self.entrypoint
 
     @property
-    def flattened_data(self):
-        self.raise_on_empty_entrypoint()
-        new_data_files = {}
-        deploy_dir = dirname(self.entrypoint) if isfile(self.entrypoint) else self.entrypoint
-        deploy_dir = self.deploy_dir or deploy_dir
+    def flattened_data(self) -> dict[str, ManifestDataFile]:
```
Comment (wch) [inline rsconnect/bundle.py:276]: Remove `@property`, rename to `get_manifest_files()`.

File: rsconnect/bundle.py [hunk 3]
```diff
@@ -283,20 +293,28 @@ def flattened_data(self):
 
     @property
     def flattened_buffer(self) -> dict[str, str]:
```
Comment (wch) [inline rsconnect/bundle.py:295]: Remove `@property`, rename to `get_manifest_files_from_buffer()`.

File: rsconnect/bundle.py [hunk 4]
```diff
@@ -306,7 +324,7 @@ def flattened_primary_html(self):
 
     @property
     def flattened_copy(self):
-        self.raise_on_empty_entrypoint()
+        self.check_and_get_entrypoint()
```
Comment (wch) [inline rsconnect/bundle.py:327]: Remove

**Review by mmarchetti — APPROVED** [review]

**Pull request merged**